### PR TITLE
RoR training step 9, implement localization, timezone setting and cus…

### PR DESCRIPTION
**Pull request by sunao-uehara**
> ステップ9: アプリの各種設定を行いましょう
日本語部分を共通化
Railsのi18nの仕組みを利用して、日本語リソースの共通化をしましょう
※i18n化を行うと後のステップでメッセージを出したりするのが楽になるなどのメリットがあります
タイムゾーンの設定
Railsのタイムゾーンを日本（東京）に設定しましょう
エラーページの設定
Railsが用意しているデフォルトのエラーページを自分が作った画面にしてみましょう
状況に応じて、適切にエラーページを設定しましょう
ステータスコードの404ページと500ページの2種類の設定は少なくとも必須とします

I completed step9. Please find attached screeenshots for localization and custom error pages.

I added locales/ja.yml to add corresponding translations. This is a screeenshot of task detail page.
ja_location_#show

These are custom 404 and 500 error pages. I created an errors_controller and generic error template to handle general http status. The controller checks URL segment /errors/{status_code} and return corresponding error message. 
custom_404
[
custom_500
](url)

One weird thing about this is if you go visit /errors/200, it shows like this as error page even though it is 200.
custom_200

And this is unsupported http status. We might consider redirect this and 20x status to 404. But I think this isn't a big deal at this moment.
custom_unsupported_code

### CODE REVIEW

File: myapp/app/controllers/errors_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class ErrorsController < ApplicationController
+  rescue_from StandardError, with: :internal_server_error
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+  def not_found
+    redirect_to "/errors/404"
```
Comment (MioArai) [inline myapp/app/controllers/errors_controller.rb]: Could you please run RuboCop once?

Installing it in your IDE might help you.
[VSCodeでRubocopを使う](https://qiita.com/d0ne1s/items/a3566aca023d11f2bfc4)
[RuboCop | RubyMine ドキュメント](https://pleiades.io/help/ruby/rubocop.html)

Comment (sunao-uehara) [reply myapp/app/controllers/errors_controller.rb]: Somehow rubocop doesn't trigger warning for quotation things on my IDE/vscode. So, I explicitly defined below in my rubocop.yml to resolve the issue.
```
StringLiterals:
  EnforcedStyle: single_quotes
```

**Review by hayashiR1 — COMMENTED** [review]
How about change these text to i18n also?

```
  MSG_CREATE_SUCCESS = 'Created a task successfully.'
  MSG_CREATE_FAILURE = 'Failed to create a task.'
  MSG_UPDATE_SUCCESS = 'Updated a task successfully.'
  MSG_UPDATE_FAILURE = 'Failed to update a task.'
  MSG_DELETE_SUCCESS = 'Deleted a task successfully.'
```

**Comment by sunao-uehara** [comment]
> How about change these text to i18n also?
> 
> ```
>   MSG_CREATE_SUCCESS = 'Created a task successfully.'
>   MSG_CREATE_FAILURE = 'Failed to create a task.'
>   MSG_UPDATE_SUCCESS = 'Updated a task successfully.'
>   MSG_UPDATE_FAILURE = 'Failed to update a task.'
>   MSG_DELETE_SUCCESS = 'Deleted a task successfully.'
> ```

@hayashiR1  Thank you for the suggestion. I added the translation for those flash messages as well.

**Review by hayashiR1 — APPROVED** [review]
LGTM !
c9dde5ac-c594-4d3c-85dd-423bf52bae77

**Pull request merged**